rules/REQUEST-921-PROTOCOL-ATTACK.conf

```diff
@@ -305,6 +305,31 @@ SecRule REQUEST_URI "@rx unix:[^|]*\|" \
     setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
+# Detection for Cookie Sandwich attacks
+# This issue affects Apache Tomcat, Python, and maybe others.
+# See:
+# - https://portswigger.net/research/stealing-httponly-cookies-with-the-cookie-sandwich-technique
+# - https://github.com/owasp-modsecurity/ModSecurity/wiki/Reference-Manual-%28v2.x%29#seccookieformat
+SecRule REQUEST_COOKIES "@rx ^\s*\$Version=1" \
```

Comment:
This will detect any CORS cookie, as `$Version` is mandatory (https://datatracker.ietf.org/doc/html/rfc2109#section-4.3.4). The sandwich is special in that it abuses quotes. You could try to detect something like `[^=]=[^"]+"`, i.e., a value that isn't quoted but **ends** with a quote (of vice versa, see the "start" and "end" params in the Portswigger post).

Reply:
🤔

Reply:
I looked at the spec again. While what I wrote above is true, `$Version` is indeed part of the original cookie syntax that has been long obsolete. You could argue that we can simply assume that most servers should never see a cookie like this, and if they do, it's probably an attack. There's a chance that very old clients may still send such cookies.

In summary, I think it's ok to simply detect the legacy cookie and block. However, I think the log message and log comment should reflect that we don't actually detect an attack but simply prevent the use of the legacy cookie format.